### PR TITLE
Changing ifconfig provider to reduce blank lines in redhat and debian ifconfigs

### PR DESCRIPTION
**Pull request by jmherbst**
Changing ifconfig provider to use '-' trim_mode for its ERB templates and updating the redhat and debian ifconfigs to leverage the trim_mode so they are not constructed with any blank lines when nil values are encountered.

This addresses https://github.com/chef/chef/issues/10457

## Description
This improves the `ifconfig` templates defined for debian and redhat so that `nil` values do not generate blank lines in the resulting ifconfig file.

Ruby ERB supports a `trim_mode` parameter of `'-'` to `"-  omit blank lines ending in -%>"`

I've changed the ERB call in https://github.com/chef/chef/blob/master/lib/chef/provider/ifconfig.rb#L239 to leverage this trim mode.

I've also changed each of the distro's template generation scripts to generate a template that omits blank lines when a `nil` value is encountered 

## Related Issue
https://github.com/chef/chef/issues/10457

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: lib/chef/provider/ifconfig/debian.rb [hunk 1]
```diff
@@ -32,25 +32,43 @@ class Debian < Chef::Provider::Ifconfig
         def initialize(new_resource, run_context)
           super(new_resource, run_context)
           @config_template = %{
-<% if new_resource.device %>
-<% if new_resource.onboot == "yes" %>auto <%= new_resource.device %><% end %>
+<% if new_resource.device -%>
```
Comment (phiggins) [inline lib/chef/provider/ifconfig/debian.rb:35]: If the whole thing depends on `new_resource.device`, that conditional could be moved out of the ERB string and just set `@config_template = ""` or something.

**Comment by phiggins** [comment]
The code looks fine, thanks for the fix! Would it be possible to add some tests for this?

**Comment by jmherbst** [comment]
What are you thinking of In addition to the existing unit tests doing string matching ([redhat](https://github.com/chef/chef/blob/master/spec/unit/provider/ifconfig/redhat_spec.rb#L57), [debian](https://github.com/chef/chef/blob/master/spec/unit/provider/ifconfig/debian_spec.rb#L70-L75)) @phiggins ?

**Comment by phiggins** [comment]
Oops, I was thinking that any existent tests would have failed with these changes, but I guess I was wrong. This seems good to me 👍

**Review by phiggins — APPROVED** [review]

**Pull request merged**